### PR TITLE
Add more error patterns

### PR DESCRIPTION
**Pull request by Binyang2014**
Add more error patterns
Enable to recognize more type of errors

### CODE REVIEW

File: src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml [hunk 1]
```diff
@@ -840,6 +848,202 @@ spec:
...
+  repro:
+  - "Running latest pytorch with old cuDNN lib"
+  solution:
+  - "Change docker image with correct DL framework version"
+  pattern:
+    runtimeContainerPatterns:
+    - userLogRegex: "dropout supported only in cudnn v"
+
+- code: 226
+  phrase: ContainerGpuOOMKilled
```
Comment (yqwang-ms) [inline src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml]: >ContainerGpuOOMKilled [](start = 10, length = 21)

ContainerCudaOOMKilled

File: src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml [hunk 2]
```diff
@@ -840,6 +848,202 @@ spec:
...
+  - "Allocate GPU memory exceed hardware limit"
+  solution:
+  - "Select GPU type which has more physical memory"
+  - "Tuning the program parameter to use less memory"
+  pattern:
+    runtimeContainerPatterns:
+    - userLogRegex: "cuda runtime error (2) : out of memory"
+
+- code: 227
+  phrase: ContainerEnginCoreDump
```
Comment (yqwang-ms) [inline src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml]: >ContainerEnginCoreDump [](start = 10, length = 22)

ContainerEngineCoreDump

File: src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml [hunk 3]
```diff
@@ -840,6 +848,202 @@ spec:
...
+  reaction: NEVER_RETRY
+  reason: "Container failed due to engine core dump"
+  repro:
+  - "Access invalid memory address"
+  solution:
+  - "Correct the user code"
+  pattern:
+    runtimeContainerPatterns:
+    - userLogRegex: "segmentation fault"
+    - userLogRegex: "core dumped"
```
Comment (yqwang-ms) [inline src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml]: Why include this into engine core dump

File: src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml [hunk 4]
```diff
@@ -804,6 +811,7 @@ spec:
   pattern:
     runtimeContainerPatterns:
     - userLogRegex: "(?msi)tensorflow.*ResourceExhaustedError.*OOM.*"
+    - userLogRegex: "ran out of memory trying to allocate"
```
Comment (yqwang-ms) [inline src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml]: This is also GPU OOM in most cases, so it is not the same kind as "ran out of memory trying to allocate"

File: src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml (hunk 3 above)
Comment (Binyang2014) [reply src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml]: According to log pattern in Philly, use log will print this when core dump occur. But don't find any clue when searching it. Remove it at this time

---
In reply to: [353553979](https://github.com/microsoft/pai/pull/3959#discussion_r353553979) [](ancestors = 353553979)

File: src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml (hunk 4 above)
Comment (yqwang-ms) [reply src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml]: Why remove "ran out of memory trying to allocate"

---
In reply to: [353554911](https://github.com/microsoft/pai/pull/3959#discussion_r353554911) [](ancestors = 353554911)

File: src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml (hunk 3 above)
Comment (yqwang-ms) [reply src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml]: ContainerEngineCoreDump is not segmentation fault

---
In reply to: [353563197](https://github.com/microsoft/pai/pull/3959#discussion_r353563197) [](ancestors = 353563197,353553979)

**Pull request merged**